tests/Functional/CustomTestKernel.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the BazingaGeocoderBundle package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Bazinga\GeocoderBundle\Tests\Functional;
+
+use Nyholm\BundleTest\TestKernel;
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Debug\DebugClassLoader as LegacyDebugClassLoader;
+use Symfony\Component\DependencyInjection\Dumper\Preloader;
+use Symfony\Component\ErrorHandler\DebugClassLoader;
+use Symfony\Component\Filesystem\Filesystem;
+
+/*
+ * Needed by PluginInteractionTest, so the test uses the cache for geoCoder, and doesn't clear it each time
+ * BUT doesn't use the cache for the service parameters, like it is happening in dev
+ *
+ * warmupDir is redefined because the method initializeContainer() is using it, but it's private in the parent.
+ * the methods using it (reboot() and getKernelParameters() and setAnnotatedClassCache() ) therefore needed to be redeclared, in order
+ * for them to have a correct value in it.
+ */
+class CustomTestKernel extends TestKernel
+{
+    private $warmupDir;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reboot(?string $warmupDir)
+    {
+        $this->shutdown();
+        $this->warmupDir = $warmupDir;
+        $this->boot();
+    }
+
+    /*
+     * Needed, otherwise the used cache is different on each kernel boot, which is a big issue in PluginInteractionTest
+     */
+    public function getCacheDir(): string
+    {
+        return realpath(sys_get_temp_dir()).'/NyholmBundleTest/cachePluginInteractionTest';
+    }
+
+    /**
+     * Returns the kernel parameters.
+     */
+    protected function getKernelParameters(): array
```

Comment:
It's the same code as in `Symfony\Component\HttpKernel\Kernel`, so why do we need it?

Reply:
I made this whole class only to modify the method initializeContainer. But this method uses the variable warmupDir, which is defined in the parent class. Unfortunately, this variable is private, thus unusable in this child class. So I had to redefine the variable. This variable is independant from the warmupDir variable of the parent, so I had to make 2 things sure :
That this child variable would have its value correctly set (through the redefining of the method reboot()) but also that the methods defined in the parent class needing this variable, would use the one of the child class, not the parent.
If I don't redefine these methods, they use warmupDir from the parent (I tested independantly). So I copy pasted these method, but they don't access the same warmupDir.